src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseTransferService.java

```diff
@@ -0,0 +1,141 @@
+@Service("caseTransferService")
+public class CaseTransferService {
+
+    private final PersistentQHelperService persistentQHelperService;
+    private final CcdClient ccdClient;
+    private static final String MESSAGE = "Failed to retrieve the case for case id : ";
+
+    @Value("${ccd_gateway_base_url}")
+    private String ccdGatewayBaseUrl;
+
+    private CaseData getOriginalCase(CaseDetails caseDetails, String userToken) {
+        try {
+            CaseData caseData = caseDetails.getCaseData();
+            if (caseData.getCounterClaim() != null && !caseData.getCounterClaim().trim().isEmpty()) {
+             List<SubmitEvent> submitEvents =  ccdClient.retrieveCasesElasticSearch(userToken, caseDetails.getCaseTypeId(), Arrays.asList(caseData.getCounterClaim()));
+            return submitEvents.get(0).getCaseData();
+            }
+            return caseDetails.getCaseData();
+        }
+         catch (Exception ex) {
+            throw new CaseCreationException(MESSAGE + caseDetails.getCaseTypeId() + ex.getMessage());
+        }
+    }
+    private List<CaseData> getAllCasesToBeTransferred(CaseDetails caseDetails, String userToken) {
+        try {
+            CaseData caseData = getOriginalCase(caseDetails, userToken);
+            List<CaseData> cases = new ArrayList<>();
+            cases.add(caseData);
+             if (caseData.getEccCases() != null && !caseData.getEccCases().isEmpty()) {
+                List<String> counterClaimList =  caseData.getEccCases().stream().map(eccCounterClaimTypeItem -> eccCounterClaimTypeItem.getValue().getCounterClaim()).collect(Collectors.toList());
+                cases.addAll(ccdClient.retrieveCasesElasticSearch(userToken,caseDetails.getCaseTypeId(),counterClaimList).stream().map(submitEvent -> submitEvent.getCaseData()).collect(Collectors.toList()));
+            }
+            return cases;
+        }
+        catch (Exception ex) {
+            throw new CaseCreationException(MESSAGE + caseDetails.getCaseTypeId() + ex.getMessage());
+        }
+    }
+
+    public void createCaseTransfer(CaseDetails caseDetails, List<String> errors, String userToken) {
+
+        List<CaseData> caseDataList = getAllCasesToBeTransferred(caseDetails, userToken);
+        for (CaseData caseData : caseDataList) {
+
+            if (!checkBfActionsCleared(caseData)) {
+                errors.add(
+                        "There are one or more open Brought Forward actions that must be cleared before the case "
+                                + caseData.getEthosCaseReference() + " can "
+                                + "be transferred");
+            }
+
+            if (!checkHearingsNotListed(caseData)) {
+                errors.add(
+                        "There are one or more hearings that have the status Listed. These must be updated before the case "
+                                + caseData.getEthosCaseReference() + " can be transferred");
+            }
+
+            if (!errors.isEmpty()) {
```

Comment:
If you have the check for errors here rather than after the for loop then the user will only see errors for a single ECC at a time. Is it better to get errors for all ECC cases? Maybe a question for a BA